### DISCUSSION TITLE
Part plans/suggestion to do

### DISCUSSION THREAD

**Discussion by likeproblem** [likeproblem/RestockRecolour#8] [Collaborator Disscusion]
VERY_COOL_PLACEHOLDER

**Comment by Spartwo** [comment]
Working on some more parts today, don't forget that ReStock variantises by changing models rather than textures so for fuel tanks you need to config and mask for each version.

**Reply by likeproblem** [reply]
I usually ignore variants unless it's something it's something massive change such as boat tail, and in the same texture file, etc

**Reply by Spartwo** [reply]
> I usually ignore variants unless it's something it's something massive change such as boat tail, and in the same texture file, etc

That's gonna cause issues when folks are wondering why it's not painting. The jumbo 64 for example has different geometry between the white and orange versions.

**Reply by Spartwo** [reply]
Marked off the Radial Nodes, Ore Tanks, and Sentinel

**Reply by Spartwo** [reply]
Remasked the drone cores to have a common pattern

**Reply by likeproblem** [reply]
> > I usually ignore variants unless it's something it's something massive change such as boat tail, and in the same texture file, etc
> 
> That's gonna cause issues when folks are wondering why it's not painting. The jumbo 64 for example has different geometry between the white and orange versions.

The issue is most variants are in another texture file, and just making another texture set for it would be quite a hassle, so I will leave variants for now, maybe an issue for post-release

**Reply by Spartwo** [reply]
Restock specifically tends to put all variants into a single texture and toggle the objects instead. It's not normal for ksp but it's very fortunate for this type of thing.

**Reply by likeproblem** [reply]
"Tends" is an understatement, almost all variants I met were in -2 file lol, check 25mTanks for example

**Comment by likeproblem** [comment]
@Spartwo Also, since I see you are working on decouplers and plates, could I ask a favour? I colored decouplers with green stripes and blue arrows, same for plates(and green bottom and blue lines), so please do the same. And if possible, can you figure out why boat tail refuses to work?

**Comment by Spartwo** [comment]
100!

**Reply by likeproblem** [reply]
![1000004393](https://github.com/user-attachments/assets/6ea4ca72-933c-4ab4-9adb-f9cb79170325)
Truly gives off this vibes

This is great news! We just started 3 weeks ago, so with this speed we can finish it by Christmas!

**Comment by Spartwo** [comment]
![image](https://github.com/user-attachments/assets/dbd376bc-4bca-4e3a-b60c-f48e7882cb41)
Remembered how to do multiple masks on the one part

**Reply by likeproblem** [reply]
How?? Variant switcher?

**Reply by Spartwo** [reply]
No there's a certain way to declare secondary materials so they don't conflict on the same textureset

MATERIAL
    {
		shader = TU/Metallic
		keyword = TU_RECOLOR

**Reply by likeproblem** [reply]
You just use different keyword?

**Reply by Spartwo** [reply]
and make sure to declare the objects, otherwise yea

**Reply by UncleMateo** [reply]
> No there's a certain way to declare secondary materials so they don't conflict on the same textureset
> 
> MATERIAL { shader = TU/Metallic keyword = TU_RECOLOR

I _knew_ there was a way to do that!  I just couldn't figure it out

**Reply by Spartwo** [reply]
![image](https://github.com/user-attachments/assets/4eff55bc-2e44-4490-acef-80352d3483bb)

Couplers marked off

![image](https://github.com/user-attachments/assets/5d53f5a0-2104-4348-bf54-284dbfa6f284)

That misalignment isn't TURD

**Comment by likeproblem** [comment]
@Spartwo do you know what you did is so revolutionary? I was **suffering** from this problem since I started the project! I think you remember when I asked ZZetho for that, he has to know this!

Also with this we can group multiple meshes with different textures under one selector without any problems! Cupola and Mainsail were ones which suffered a lot from that! You are awesome, thank you so much! ❤️

**Comment by EmberSkyMedia** [comment]
I'm working on the relays (coloured the stacks tonight, working on the deploying tomorrow and then will ensure they are work in game and push the changes to the github fork).
![image](https://github.com/user-attachments/assets/db920aa8-c0f8-4f42-a4bc-5c7dc61eb45a)

**Reply by Spartwo** [reply]
Nice pattern consistency with the orbital scanner so far

**Reply by EmberSkyMedia** [reply]
Yes, I’m looking at previously done parts and trying to match the same style.

**Comment by Spartwo** [comment]
Comms 16 antennas done with consolidation of the image asset files. The different parts are segregated into thier own colour layers.

Will get the engine plates done before moving onto the struts or solar parts

**Reply by Spartwo** [reply]
Also started turning the restock models folder into temporary housing for fixes. Thankfully we're almost at 50% and still only 2 parts need changed

**Reply by likeproblem** [reply]
When you get engine plates please look why boattails don't work. I'll probably work on fuel tank variants today, a lot of them piled up

**Reply by Spartwo** [reply]
Working fine for me, I've rewritten the structure to do all of them so maybe it was however it was configured

**Comment by likeproblem** [comment]
@Spartwo why did you remove shrouds from engine plates?

**Reply by likeproblem** [reply]
Nvm you just did wizardry with excludeMesh, though why did you exclude fairing from texture switch?? All parts should have separate switch for fairings

**Reply by Spartwo** [reply]
Personal preference but sure I'll swap em back to multiple switchers

**Reply by likeproblem** [reply]
Don't, im commiting them rn

**Comment by likeproblem** [comment]
@Spartwo something happened to structural tubes... 5m had those blue tubes/pipes/structures inside, what happened?
![1729970601439134380235954368073](https://github.com/user-attachments/assets/deeac40a-3ac8-4f62-9597-7f7bc6c84fcc)
Also seems like only engine plate 25m is colored, I guess because it was just a placeholder

**Reply by Spartwo** [reply]
There's two versions, one is blue one is unpainted

**Reply by likeproblem** [reply]
How do I switch between them?

**Reply by Spartwo** [reply]
Orange variants have painted ends and trusses.

**Reply by Spartwo** [reply]
Only the engine plate 2.5m is painted because the config is there but the mask isn't yet

**Reply by likeproblem** [reply]
> Orange variants have painted ends and trusses.

Ooh, interesting approach!

> Only the engine plate 2.5m is painted because the config is there but the mask isn't yet

I figured out

**Comment by EmberSkyMedia** [comment]
Is someone doing the  remaining non-relay antennas? Or should I colour those next?

Alt: I could do the ISRU or Ladders 🪜

**Reply by likeproblem** [reply]
I think ladders are simple flat colour, defaultpaint. I think nobody works on non relay right now, except I need to implement your png into actual config. Isru would be appreciated too

**Reply by Spartwo** [reply]
Half the non relays are already done but nobody has claimed the others

**Reply by EmberSkyMedia** [reply]
Okay, I'll do the remaining antennas, then move to ladders (easy/quick), and then the ISRU parts in Utilities

**Reply by likeproblem** [reply]
Can I ask what you would do with ladders? Aren't they just simple grey texture?

**Reply by EmberSkyMedia** [reply]
Likely just a blanket colour or maybe have the rungs one colour and the bars another… I thought about stripes but not sure that’s something people would want.

![image](https://github.com/user-attachments/assets/e926927e-de9f-417e-85f6-902fc7fc81c9)

**Reply by Spartwo** [reply]
Restock ladders have detail

**Reply by likeproblem** [reply]
What about this? Mostly red but green in middle?
![markup_1000004473](https://github.com/user-attachments/assets/34d41334-5332-415f-b0d8-fef8fdd83fed)

**Reply by Spartwo** [reply]
![image](https://github.com/user-attachments/assets/0e614f5c-9c75-4158-b240-a0d23def5d90)
There's an underlying pattern in the base texture mostly to that effect. 
![image](https://github.com/user-attachments/assets/0eed293a-15ef-448f-94c0-09dead757bc0)

**Reply by EmberSkyMedia** [reply]
Okay, I can do something like that.

**Comment by EmberSkyMedia** [comment]
Ladders and Antenna Paints are Done (Waiting on likeproblem configs).
I'm busy the next 2-3 days but I'll start on any remaining "RESOURCE" Folder items next (currently it looks like ISRU & Drill).

**Reply by Spartwo** [reply]
I've done the surface scanner

**Reply by EmberSkyMedia** [reply]
Okay, scratch scanners as they are both done :)

**Reply by likeproblem** [reply]
What are the scratch scanners again?

**Comment by UncleMateo** [comment]
I'm still having problems with these dang claws.. I had one of them more-or-less working, but I broke it in trying to fix its issues, but now I can't even get back to that point. but I've decided to just put work into another part whenever I get frustrated, so as to bring my rate of contribution higher when I get stuck like this (which seems to be my specialty).  You'll see I just did that, just finished the larger structural hub by Rockomax. 
In all honesty I was kind of demoralized at the loss of the forums, making it harder to continue the Sisyphean task of trying to figure out problematic parts.  But things are looking up! The forums aint dead, at least not yet!

**Reply by likeproblem** [reply]
Can you commit whatever you have? I will check it and maybe try to fix it

**Comment by Spartwo** [comment]
![image](https://github.com/user-attachments/assets/2ffea119-77cb-47a5-8a5f-8888bc2b7cd2)

**Reply by likeproblem** [reply]
Engine plates?

**Reply by Spartwo** [reply]
and boat-tails

**Reply by likeproblem** [reply]
Thanks :)
Again, sorry I can't really work much, but I hope to get at least 10 parts get done over weekends

**Reply by Spartwo** [reply]
I'm not exactly blasting through these myself right now

**Reply by UncleMateo** [reply]
Lookin' good!  I'll try and put up a few parts tonight.  I claimed a bunch, might take me a couple of days.  The way it's looking, we might start crossing off several categories in short order, but certainly not fuel tanks, and _definitely_ not engines.  Engines are gonna be the end boss of this pack, the way things are shaping up, and it makes sense.. each one is generally a complicated texture file, a lot of them have multiple variants with different meshes, etc.  I think getting that little check-mark up is good for morale, but after making a third list in notepad, only including the parts which we hadn't done yet, it became apparent that I need to start doing more engines.  So from now on, I will always try to add an engine or two with whatever I'm working on.

Here's the breakdown, this counter includes all the Restock & Restock+ parts of course, but also included are any of the Stock/BG/MH parts that _were not_ a part of Restock which did not receive full recolors, or which did receive some recolors that are broken by either Restock or RestockRecolour.  In other words, this is the maximized part count right now.  And, this was something I did a few days ago so I don't think it included the last update Spartwo did. 

**parts left**
* engine:36
* fuel_tnk:25
* utility:18
* groud:15
* aeor:12
* coupling:12
* payload:12
* electrical:11
* thermal:11
* structural:10
* command:9

**Comment by Spartwo** [comment]
![image](https://github.com/user-attachments/assets/f89e8b86-b4f9-43a5-a59e-bd9ae222349d)

_Pamnel_

**Reply by Spartwo** [reply]
I did promise to do certain chunks of parts so I'm following through on that still

**Reply by Spartwo** [reply]
Electrical category is done

**Reply by UncleMateo** [reply]
Oh man, you made my day!  I appreciate you very much.  You breathed a little more life into my resolve for completing this.  Your timing is great, you've given me a surge of kerbal morale.

I'm afraid to say I did the fuel release valve part myself, I think you had initially claimed that one.. I was at first really trying to respect the 'part association list' but I eventually stopped checking it quite so much and forgot what was on there, I didn't mean to be rude or anything it was just an easy looking part and I needed one like that to increase my TURD morale so I could tackle the more complicated issues.  I kind of forgot it was spoken for.

Also, I just want to thank you for all the work you put into this project.  In respect of your efforts I will try and see things through til the end.

**Reply by Spartwo** [reply]
I hadn't claimed the valve, only the fuel line

**Reply by UncleMateo** [reply]
Okay, cool!  Well hopefully I didn't rudely steal a different part from you.  But if so, know there was no ill will involved.

**Reply by Spartwo** [reply]
![Image](https://github.com/user-attachments/assets/ffb415e8-b59e-41fb-8b85-112b23063236)

_Strutting My Stuff_

**Reply by Spartwo** [reply]
![Image](https://github.com/user-attachments/assets/e05c8b28-c21e-4b55-9d51-971833bc365a)

*Looking Radiant*

**Reply by Spartwo** [reply]
![image](https://github.com/user-attachments/assets/c1c09f6f-fa34-4571-9286-995b2cbd771a)

**Reply by Spartwo** [reply]
![image](https://github.com/user-attachments/assets/d50f62d2-9852-4cbb-af71-c35a7df30e8c)

![image](https://github.com/user-attachments/assets/38d4e170-dc61-4e99-ae38-5037561fa60f)

**Comment by EmberSkyMedia** [comment]
With the release of Firefly I have a bit more time this week to do a few parts, what needs to get done?

**Reply by Spartwo** [reply]
The bulk of stuff left is in fuel tanks and engines. Especially the latter.

**Reply by EmberSkyMedia** [reply]
Do we have a plan for the engines or a guide?
My thinking (if I may) is whatever we do to Restock to mirror CyroEngines and Extensions.

**Reply by Spartwo** [reply]
I don't believe enough of them are done to have a colour pattern that needs to yet be conformed with.

**Reply by EmberSkyMedia** [reply]
I’ll have a look at stock and missing history and see how they did it, ideally the RGB is such that colours look reasonably uniform.

**Reply by EmberSkyMedia** [reply]
I’ll start with the solid fuel rockets if no one has objections.

**Comment by Spartwo** [comment]
![image](https://github.com/user-attachments/assets/a5292a60-c9c3-42fa-9cf8-1a4deccca346)


Commit 200 and 69%

**Comment by Spartwo** [comment]
How we feeling now restock pbr has been a nothingburger?

**Reply by EmberSkyMedia** [reply]
I haven't heard a peep about it.

**Reply by UncleMateo** [reply]
I saw this message in the form of an email from GitHub and it is the thing that inspired me to get back to work on RestockRecolour, just so you know!  I guess you could say I'm feeling inspired!
